src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ServerInfo.cs

```diff
@@ -64,5 +68,23 @@ public class ServerInfo
         /// The Operating System version string of the machine running the SQL Server instance.
         /// </summary>
         public string MachineName { get; set; }
+
+        /// <summary>
+        /// Server options
+        /// </summary>
+        public Dictionary<string, object> Options { get; set; }
+    }
+}
+
+public class BigDataClusterEndpoint
+{
+    public string ServiceName;
```

Comment:
The pattern is to use getters / setters for all fields in JSON-serializable contract classes. I'd also suggest following standard practice to not have a constructor, and just do
```c#
new BigDataClusterEndpoint() { ServiceName = 'xyz', ... }
```
to assign to setters during construction instead